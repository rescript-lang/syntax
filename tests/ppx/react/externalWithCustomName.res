@@jsxConfig({version: 3})

module Foo = {
  @react.component @module("Foo")
  external component: (~a: int, ~b: string, _) => React.element = "component"
}

let t = <Foo.component a=1 b={"1"} />

@@jsxConfig({version: 4, mode: "classic"})

module Foo = {
  @react.component @module("Foo")
  external component: (~a: int, ~b: string, _) => React.element = "component"
}

let t = <Foo.component a=1 b={"1"} />

@@jsxConfig({version: 4, mode: "automatic"})

module Foo = {
  @react.component @module("Foo")
  external component: (~a: int, ~b: string, _) => React.element = "component"
}

let t = <Foo.component a=1 b={"1"} />
