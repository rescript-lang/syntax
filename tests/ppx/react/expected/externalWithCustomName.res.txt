@@jsxConfig({version: 3})

module Foo = {
  @obj
  external componentProps: (~a: int, ~b: string, ~key: string=?, unit) => {"a": int, "b": string} =
    ""
  @module("Foo")
  external component: React.componentLike<{"a": int, "b": string}, React.element> = "component"
}

let t = React.createElement(Foo.component, Foo.componentProps(~a=1, ~b={"1"}, ()))

@@jsxConfig({version: 4, mode: "classic"})

module Foo = {
  type props<'a, 'b> = {a: 'a, b: 'b}

  @module("Foo")
  external component: React.componentLike<props<int, string>, React.element> = "component"
}

let t = React.createElement(Foo.component, {a: 1, b: "1"})

@@jsxConfig({version: 4, mode: "automatic"})

module Foo = {
  type props<'a, 'b> = {a: 'a, b: 'b}

  @module("Foo")
  external component: React.componentLike<props<int, string>, React.element> = "component"
}

let t = React.jsx(Foo.component, {a: 1, b: "1"})
